chirp/drivers/baofeng_uv17Pro.py

```diff
@@ -0,0 +1,1427 @@
+                                                    _mem.vfo.a.rxtone)))
+        rs.set_apply_callback(apply_rxtone, _mem.vfo.a)
+        work.append(rs)
+
+        rs = RadioSetting("vfo.a.txtone", "VFA A TX QT/DQT",
+                          RadioSettingValueList(self.RXTX_CODES,
+                                                getToneIndex(
+                                                    _mem.vfo.a.txtone)))
+        rs.set_apply_callback(apply_txtone, _mem.vfo.a)
+        work.append(rs)
+
+        val1b = RadioSettingValueString(0, 10,
+                                        convert_bytes_to_freq(
+                                            _mem.vfo.b.freq))
+        val1b.set_validate_callback(my_validate)
+        rs = RadioSetting("vfo.b.freq", "VFO B Frequency", val1b)
+        rs.set_apply_callback(apply_freq, _mem.vfo.b)
+        work.append(rs)
+
+        rs = RadioSetting("vfo.b.sftd", "VFO B Offset dir",
+                          RadioSettingValueList(
+                              LIST_SHIFTD, LIST_SHIFTD[_mem.vfo.b.sftd]))
+        work.append(rs)
+
+        val1b = RadioSettingValueString(
+                    0, 10, convert_bytes_to_offset(_mem.vfo.b.offset))
+        rs = RadioSetting("vfo.b.offset",
+                          "VFO B Offset", val1b)
+        rs.set_apply_callback(apply_offset, _mem.vfo.b)
+        work.append(rs)
+
+        rs = RadioSetting("vfo.b.lowpower", "VFO B Power",
+                          RadioSettingValueList(
+                                LIST_TXPOWER,
+                                LIST_TXPOWER[min(_mem.vfo.b.lowpower, 0x02)]
+                                            ))
+        work.append(rs)
+
+        rs = RadioSetting("vfo.b.wide", "VFO B Bandwidth",
+                          RadioSettingValueList(
+                              LIST_BANDWIDTH,
+                              LIST_BANDWIDTH[_mem.vfo.b.wide]))
+        work.append(rs)
+
+        rs = RadioSetting("vfo.b.scode", "VFO B S-CODE",
+                          RadioSettingValueList(
+                              LIST_SCODE,
+                              LIST_SCODE[_mem.vfo.b.scode]))
+        work.append(rs)
+
+        rs = RadioSetting("vfo.b.step", "VFO B Tuning Step",
+                          RadioSettingValueList(
+                              LIST_STEP, LIST_STEP[_mem.vfo.b.step]))
+        work.append(rs)
+
+        rs = RadioSetting("vfo.b.fhss", "VFO B FHSS",
+                          RadioSettingValueBoolean(_mem.vfo.b.fhss))
+        work.append(rs)
+
+        rs = RadioSetting("vfo.b.rxtone", "VFA B RX QT/DQT",
+                          RadioSettingValueList(self.RXTX_CODES,
+                                                getToneIndex(
+                                                    _mem.vfo.b.rxtone)))
+        rs.set_apply_callback(apply_rxtone, _mem.vfo.b)
+        work.append(rs)
+
+        rs = RadioSetting("vfo.b.txtone", "VFA B TX QT/DQT",
```

Comment:
Seeing all this - shouldn't you expose the VFO memories as special channels so they show up in the spreadsheet instead of as a setting? I know some radios do it this way (probably incorrectly) but you're reimplementing a *ton* of logic here which is sort of free if these are exposed as memories.

Reply:
They implemented the frequency mode settings, slightly different in the internal memory of the radio. I don't understand why they did it this way. The frequency mode entries differ too much from the normal memory entries to treat them the same way.

Reply:
No, because you don't have to implement it with the same code. You can have different code for different-formatted memories. The FT450 driver, for example, has to do quite a bit of different stuff for the VFO channels:

See this for example:

https://github.com/kk7ds/chirp/blob/master/chirp/drivers/ft450d.py#L628

..or as above, just don't expose the VFOs at all because I don't think there's much point.